### ISSUE TITLE
[PR Reference] force fp32 logits

### ISSUE THREAD

**Issue by justusmattern27** [Proximal-Labs/prime-rl-clone-5#44]
*This is a reference to original PR #664 by @samsja*

**Original PR: https://github.com/PrimeIntellect-ai/prime-rl/pull/664**
**Status: closed**
**Created at: 2025-08-01 15:39:18+00:00**
**Merged: True**
